src/ConnectedMode/Migration/MigrationChecker.cs

```diff
@@ -24,66 +24,71 @@
 using SonarLint.VisualStudio.Core.Binding;
 using Microsoft.VisualStudio.Threading;
 
+using Task = System.Threading.Tasks.Task;
+
 namespace SonarLint.VisualStudio.ConnectedMode.Migration
 {
     [Export(typeof(MigrationChecker))]
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal sealed class MigrationChecker : IDisposable
     {
         private readonly IActiveSolutionTracker activeSolutionTracker;
-        private readonly IMigrationPrompt migrationPrompt;
+        private readonly IMefFactory mefFactory;
         private readonly IConfigurationProvider configurationProvider;
         private readonly IObsoleteConfigurationProvider obsoleteConfigurationProvider;
+        private IMigrationPrompt migrationPrompt;
 
         [ImportingConstructor]
         public MigrationChecker(
             IActiveSolutionTracker activeSolutionTracker,
-            IMigrationPrompt migrationPrompt,
+            IMefFactory mefFactory,
             IConfigurationProvider configurationProvider,
             IObsoleteConfigurationProvider obsoleteConfigurationProvider)
         {
             this.activeSolutionTracker = activeSolutionTracker;
-            this.migrationPrompt = migrationPrompt;
+            this.mefFactory = mefFactory;
             this.configurationProvider = configurationProvider;
             this.obsoleteConfigurationProvider = obsoleteConfigurationProvider;
 
             activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
 
             // Initial check incase event was fired before we registered.
-            DisplayMigrationPromptIfMigrationIsNeeded();
+            DisplayMigrationPromptIfMigrationIsNeededAsync().Forget();
```

Comment:
The threading cookbook says we should not switch threads in an ImportingConstructor however DisplayMigrationPromptIfMigrationIsNeededAsync() maybe does switch but not sure if the Forget() is good enough or not

Reply:
Good question. Might be worth trying the suggested [code snippet](https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md#how-do-i-effectively-verify-that-my-code-is-fully-free-threaded) that would force deadlocks to surface.

Reply:
Actually, one simple way around this would be to make the package that creates the migration checker responsible for triggering the initial check (i.e. add a public method to the interface and have the package call it).